packages/api/tests/lambdas/test-report-executions.js

```diff
@@ -0,0 +1,210 @@
+'use strict';
+
+const test = require('ava');
+
+const { getExecutionArn } = require('@cumulus/common/aws');
+const { randomId, randomNumber, randomString } = require('@cumulus/common/test-utils');
+
+const Execution = require('../../models/executions');
+const { getReportExecutionMessages, handler } = require('../../lambdas/report-executions');
+
+const fakeExecutionRecord = {
+  name: randomString(),
+  arn: randomString(),
+  collectionId: randomId('collection'),
+  status: 'completed',
+  execution: randomString(),
+  error: {
+    Error: 'Error',
+    Cause: 'Workflow failed'
+  },
+  createdAt: Date.now(),
+  duration: 10000000,
+  timestamp: Date.now() - randomNumber(10000000)
+};
+
+const createFakeExecutionRecord = (granuleParams) => ({
+  ...fakeExecutionRecord,
+  ...granuleParams
+});
+
+const createExecutionSnsMessage = (message) => ({
+  EventSource: 'aws:sns',
+  Sns: {
+    Message: JSON.stringify(message)
+  }
+});
+
+let executionsModel;
+
+test.before(async () => {
+  process.env.ExecutionsTable = randomId('executionsTable');
+  executionsModel = new Execution();
+  await executionsModel.createTable();
+});
+
+test.beforeEach((t) => {
+  t.context.stateMachine = randomId('stateMachine');
+  t.context.executionName = randomString();
+  t.context.arn = getExecutionArn(t.context.stateMachine, t.context.executionName);
+  t.context.createdAtTime = Date.now();
+  t.context.collectionId = randomId('collection');
+  t.context.originalPayload = {
+    foo: 'bar'
+  };
+  t.context.finalPayload = {
+    test: randomString()
+  };
+});
+
+test.after.always(async () => {
+  await executionsModel.deleteTable();
+});
+
+test('getReportExecutionMessages returns correct number of messages', (t) => {
```

Comment:
Not sure if this test is necessary. All it is proving is that our filtering helper isSnsEvent works properly, which has its own separate unit tests. It was more useful before the current refactor when this lambda received full execution messages and we wanted to verify that they were valid.